lib/bootsnap/load_path_cache.rb

```diff
@@ -54,7 +54,10 @@ def unload!
       end
 
       def supported?
-        RUBY_ENGINE == "ruby" &&
+        # https://github.com/oracle/truffleruby/issues/3131
+        return false if RUBY_ENGINE == "truffleruby" && RUBY_ENGINE_VERSION < "23.1.0"
+
+        %w[ruby truffleruby].include?(RUBY_ENGINE) &&
           RUBY_PLATFORM =~ /darwin|linux|bsd|mswin|mingw|cygwin/
```

Comment:
this would remove the PLATFORM restriction when the engine is truffleruby and i'm not sure we want that.

we could do something like

```suggestion
        case RUBY_ENGINE
        when "truffleruby"
          # https://github.com/oracle/truffleruby/issues/3131
          return false if RUBY_ENGINE_VERSION < "23.1.0"
        when "ruby"
          # fall through
        else
          return false
        end

        RUBY_PLATFORM =~ /darwin|linux|bsd|mswin|mingw|cygwin/
```

Reply:
Right I missed that it was applying to both. if so, let's do

```ruby
if RUBY_PLATFORM.match?(/darwin|linux|bsd|mswin|mingw|cygwin/)
  case RUBY_ENGINE
  # ...
end
```

Reply:
Done.